### ISSUE TITLE
Mechanism to deploy HIL in an environment which is already in production

### ISSUE THREAD

**Issue by naved001** [CCI-MOC/hil#801]
This came up while I was having a discussion with @henn about upgrading HaaS in one of our environments that do no support migrations. The problem is we already have some stuff that is configured (switch has certain vlans enabled on certain ports). We can do the following: 

1. Tell admins to suck it up, start from scratch and then configure everything again using HIL. People may not be too happy about it as it would lead to some downtime.

2. Have an API to import switch configuration (so, port and vlan related info) and then populate the HaaS database to be in sync with the switch. But this may not work, because HaaS database has more information like project names, network names etc which we cannot get from the switch. 

3. Run HaaS commands to build it's database and allow it to talk to switch. The issue here is we don't know how the switch will behave when we repeat the same commands to it. (Eg; a port already has some vlan set, and then we do it again using HIL). 

4. Run HaaS commands in a "database only" mode that does not actually talk to the switch, but only builds HaaS' database. This may lead to inconsistencies in the database if any commands were to fail on a real switch.

@henn I don't know if I phrased everything right, so feel free to edit/add more stuff to make it easier to understand.

Curious about your thoughts on this @zenhack @SahilTikale @radonm

**Comment by zenhack** [comment]
I don't think it makes any sense to introduce changes into HIL proper to deal with a setup using a particular old version -- we have an upgrade path now; I don't want to add stuff that will only ever be used this once. If you guys want to avoid the downtime, it may involve some custom hackery. At that point you guys could just write some icky script that injects the necessary database contents -- this is likely the only time anyone will need to do this.

**Comment by naved001** [comment]
The issue is a bit general rather than just upgrading an old version that cannot be upgraded. The question is how do we introduce HIL to an environment where there's a bunch of stuff already configured. Unless, of course, we make it clear to new HIL users that you have to get rid of your existing config in order to use HIL.

**Comment by zenhack** [comment]
Ok, so I'd like some more fleshed-out use cases. I can see the utility of being able to slide HIL under an existing setup without disturbing anything, but I want to figure out what our scope is.

**Comment by naved001** [comment]
> able to slide HIL under an existing setup without disturbing anything

Yeah, that's a better way to phrase it.

> I want to figure out what our scope is.

I have to figure that out too. Let me think more about it.
